Source/DafnyCore/Triggers/TriggerExtensions.cs

```diff
@@ -154,7 +154,6 @@ internal static class ExprExtensions {
     private static bool SameBoundVar(IVariable arg1, IVariable arg2) {
       return arg1 == arg2 ||
              (arg1.Name == arg2.Name &&
-              arg1.CompileName == arg2.CompileName &&
```

Comment:
Is this safe?

Reply:
Yes it's a weird comparison. Is it because cloned variables have the same compile name perhaps? Anyway, comparing a unique name should be enough of a guarantee indeed.